sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/exemplar/ExemplarReservoir.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.exemplar;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.metrics.data.Exemplar;
+import java.util.List;
+
+/**
+ * An interface for an exemplar reservoir of samples.
+ *
+ * <p>This represents a reservoir for a specific "point" of metric data.
+ */
+public interface ExemplarReservoir {
+
+  /** An exemplar reservoir that stores no exemplars. */
+  public static ExemplarReservoir noSamples() {
+    return NoExemplarReservoir.INSTANCE;
+  }
+
+  /** Offers a {@code long} measurement to be sampled. */
+  void offerMeasurement(long value, Attributes attributes, Context context);
+
+  /** Offers a {@code double} measurement to be sampled. */
+  void offerMeasurement(double value, Attributes attributes, Context context);
+
+  /**
+   * Builds (an immutable) list of Exemplars for exporting from the current reservoir.
+   *
+   * <p>Additionally, clears the reservoir for the next sampling period.
+   *
+   * @param pointAttributes the {@link Attributes} associated with the metric point. {@link
+   *     Exemplar}s should filter these out of their final data state.
```

Comment:
Whose responsibility is the filtering? Is this comment intended to be simply informative, or is this directing the Reservoir implementor to do the filtering? We should definitely make it clear if this is something that implementor needs to worry about or not.

Reply:
The implementation needs to do it, added a more descriptive comment here.

Reply:
This looks the same as it was before to me. Missed push, perhaps?

Reply:
oh, I see. It's below in the `@return`. How important is this? It *sounds* like it's very important, but I think it would be easy to miss this requirement as currently written.